### PR TITLE
Testing Connection to Servers - New Post

### LINKED ISSUE #21: Request - How to get all the alive servers in the domain?
# Summary of the request
As an admin, have to do a lot of reporting for the servers in the domain, however not all the servers are always running.
Maybe using the ActiveDirectory module, get all the servers, and then do the Test-Connection command on the server to check if it is reachable, and return the alive servers.

- Active Directory Reference: https://docs.microsoft.com/en-us/powershell/module/addsadministration/get-adcomputer?view=win10-ps
- Test-Connection Reference: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/test-connection?view=powershell-7.1

### PR DESCRIPTION
**Pull request by doctordns**
# PR Summary

Fixes #21 

## PR Checklist

- Do you want this post to be published on a specific date?: Y/N [Specify date]:
- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] Includes content related an open issue
  - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

**Comment by doctordns** [comment]
This post is for next week on the basis I provide one a week!

**Review by pawanadubey — APPROVED** [review]
This looks good to me.

File: Posts/2021/04/get-live-servers.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+---
+post_title: Testing The Connection to Computers in the Active Directory
```
Comment (sdwheeler) [inline Posts/2021/04/get-live-servers.md:2]: MS Style Guide - Use sentence case not Title case
```suggestion
post_title: Testing the connection to computers in the Active Directory
```

File: Posts/2021/04/get-live-servers.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+**A:**  Of course you can do this with PowerShell! You can use the Active Directory cmdlets and `Test-Connection`, although it is not as simple as one might like.
+
+## Using the `ActiveDirectory` module
+
+Microsoft has developed several modules to help you deploy and manage AD in your organisation or via Azure.
+The `ActiveDirectory` module is one which Microsoft ships with Windows Server (although not installed by default).
+You can also load the Remote Server Administration (RSAT) module for AD on a Windows 10 host.
+The RSAT module allows you to manage the AD using PowerShell from a remote machine.
+For more details on the `ActiveDirectory` module, see the [ActiveDirectory module documentation](https://docs.microsoft.com/powershell/module/addsadministration/().)
```
Comment (sdwheeler) [inline Posts/2021/04/get-live-servers.md:20]:
```suggestion
For more details on the `ActiveDirectory` module, see the [ActiveDirectory](https://docs.microsoft.com/powershell/module/addsadministration/) module documentation.
```

File: Posts/2021/04/get-live-servers.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+You can use the `Get-ADComputer` account to return details about some or all computers within the AD.
+You have several ways to use `Get-ADComputer` to get just the computer accounts you want with any property you need.
```
Comment (sdwheeler) [inline Posts/2021/04/get-live-servers.md:23]: Make imperative.

```suggestion
Use the `Get-ADComputer` account to return details about some or all computers within the AD.
There are several ways to use `Get-ADComputer` to get just the computer accounts you want with any property you need.
```

File: Posts/2021/04/get-live-servers.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+These include using the **-Identity** and **-Filter** parameters.
```
Comment (sdwheeler) [inline Posts/2021/04/get-live-servers.md:24]: See the PowerShell style guide. When referencing parameters with the dash character, use backticks. But when talking about parameters in a paragraph, use bold without the dash. You should only use the dash form when you are specifically talking about syntax.

```suggestion
These include using the **Identity** and **Filter** parameters.  
```

File: Posts/2021/04/get-live-servers.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+You can use the `Get-ADComputer` account to return details about some or all computers within the AD.
+You have several ways to use `Get-ADComputer` to get just the computer accounts you want with any property you need.
+These include using the **-Identity** and **-Filter** parameters.  
+Every computer account returned by `Get-ADComputer` contains two important properties: **Name** and **DNSHostName**.
+The **Name** property is the single-label name of the computer (aka the NetBIOS name).
+The **DNSHostName** property is the fully qualified DNS name for the computer.
+Like this:
+
+```powershell-console
+PS C:\Foo> Get-ADComputer -Filter * | Format-Table -Property Name, DNSHostName
```
Comment (sdwheeler) [inline Posts/2021/04/get-live-servers.md:31]: For examples, use the simplified prompt or no prompt at all unless the path displayed is relevant to the example. This is more platform agnostic. It also avoids the backslash bug we have with the WordPress import.

```suggestion
PS> Get-ADComputer -Filter * | Format-Table -Property Name, DNSHostName
```

File: Posts/2021/04/get-live-servers.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+What we have here is a classic, albeit relatively uncommon, situation.
+The `Test-Connection` cmdlet uses the parameter name **Target** to indicate the computer to which you are testing a connection.
+However, in this pipelined command, the objects produced by `Get-ADComputer` do not contain properties of that name.
+Instead, these objects have properties named **Name** and **DNSHostName**.
+
+Please note: with Windows PowerShell, you used the parameter ComputerName to indicate the computer you are investigating.
+With PowerShell 7, the developers have changed this parameter name `TargetName`.
+For best compatibility, the cmdlet defines the `ComputerName` alias to this parameter.
+This cmdlet lets you use either **-TargetName** or **-Computername** with `Test-Connection`.
```
Comment (sdwheeler) [inline Posts/2021/04/get-live-servers.md:69]: Use WordPress NOTE syntax and clean up formatting

```suggestion
[alert type="note" heading="Note"]With Windows PowerShell, you used the parameter **ComputerName** to indicate the computer you are investigating.
With PowerShell 7, the developers have changed this parameter name to **TargetName**.
For best compatibility, the cmdlet defines the**`ComputerName** alias to this parameter.
This cmdlet lets you use either **TargetName** or **Computername** with `Test-Connection`.[/alert]
```

File: Posts/2021/04/get-live-servers.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+## ForEach-Object to the rescue
+
+It is pretty easy to get around this parameter/property alignment challenge.
+You use the `Foreach-Object` cmdlet, like this:
+
+```powershell-console
+PS C:\Foo> Get-ADComputer -Filter * |
```
Comment (sdwheeler) [inline Posts/2021/04/get-live-servers.md:77]:
```suggestion
PS> Get-ADComputer -Filter * | 
```

File: Posts/2021/04/get-live-servers.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+          <Name>TargetName</Name>
+          <ReferencedMemberName>DNSHostName</ReferencedMemberName>
+         </AliasProperty>
+    </Members>
+  </Type>
+
+</Types>
+
+PS C:\Foo> Update-TypeData -PrependPath .\aaatypes.types.ps1xml
+PS C:\Foo> Get-ADComputer -Identity Cookham1 | Test-Connection
```
Comment (sdwheeler) [inline Posts/2021/04/get-live-servers.md:134]:
```suggestion
PS> Update-TypeData -PrependPath .\aaatypes.types.ps1xml
PS> Get-ADComputer -Identity Cookham1 | Test-Connection
```

File: Posts/2021/04/get-live-servers.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+You can persist this ETS extension by adding the `Update-TypeData` to your PowerShell profile.
+That way, every time you start a PowerShell session, that ETS extension is in place and ready to assist you.
+
+For details of and background to the ETS, see the [Extended Type System Overview](https://docs.microsoft.com/powershell/scripting/developer/ets/overview).
+
+## Summary
+
+The `Get-ADComputer` cmdlet produces objects whose properties the object developers have not aligned, pipeline wise, with Test-Connection.
+There is a simple way around that, using `For-EachObject`, although it takes a bit more typing.
+You can also use the ETS to extend the `ADComputer` object to have a more friendly alias.
```
Comment (sdwheeler) [inline Posts/2021/04/get-live-servers.md:156]: Types in bold
```suggestion
You can also use the ETS to extend the **ADComputer** object to have a more friendly alias.
```

**Review by sdwheeler — CHANGES_REQUESTED** [review]

**Comment by doctordns** [comment]
Thanks @sdwheeler for the review. I think I have all the points resolved.

**Comment by doctordns** [comment]
Given that this post is not meant to go live till next week, is there an easy way to schedule it?

**Review by sdwheeler — APPROVED** [review]

**Comment by sdwheeler** [comment]
I can set a publish date on it in WordPress.

**Comment by sdwheeler** [comment]
Scheduled for: Apr 2, 2021 at 05:00 PST

**Pull request merged**